app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperOnboarding.kt

```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.wallpapers
+
+import android.graphics.Bitmap
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.Icon
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.settings.wallpaper.WallpaperThumbnails
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * A view that shows content of a WallpaperOnboarding dialog.
+ *
+ * @param wallpapers Wallpapers to add to grid.
+ * @param currentWallpaper The currently selected wallpaper.
+ * @param onCloseClicked Callback for when the close button is clicked.
+ * @param onButtonClicked Callback for when the bottom text button is clicked.
+ * @param loadWallpaperResource Callback to handle loading a wallpaper bitmap. Only optional in the default case.
+ * @param onSelectWallpaper Callback for when a new wallpaper is selected.
+ */
+
+@Suppress("LongParameterList")
+@ExperimentalMaterialApi
+@Composable
+fun WallpaperOnboarding(
+    wallpapers: List<Wallpaper>,
+    currentWallpaper: Wallpaper,
+    onCloseClicked: () -> Unit,
+    onBottomButtonClicked: () -> Unit,
```

Comment:
```suggestion
    onExploreMoreButtonClicked: () -> Unit,
```
We could always be more descriptive with our naming here.

Reply:
but in that case we create coupling between a string resource and the parameter naming. it might go out of sync too easily

Reply:
I think this is an assumed risk for the benefit of having clear documentation/APIs. `InactiveTabsList` has similar lamdbas that are directly tied to their use cases & user-facing descriptions (e.g. `onDeleteAllButtonClick`, `onAutoCloseDismissClick`, etc.). Worst case, the lambda's name gets refactored if the button's text is updated and significantly deviates from the original wording.

Reply:
okay, three versus one, democracy, changing to `onExploreMoreButtonClicked`